src/systems/physics/Physics.cc

```diff
@@ -190,6 +191,21 @@ class ignition::gazebo::systems::PhysicsPrivate
   public: physics::FrameData3d LinkFrameDataAtOffset(
       const LinkPtrType &_link, const math::Pose3d &_pose) const;
 
+  /// \brief Get the top level model of an entity
+  /// \param[in] _entity Input entity
+  /// \param[in] _ecm Constant reference to ECM.
+  /// \return Entity of top level model
+  public: ignition::gazebo::Entity TopLevelModel(const Entity &_entity,
```

Comment:
I think these functions could be in `Util.hh`, they should be useful to other systems. Then we can also write some unit tests for them.

Reply:
moved to Util class. a71a3d2e2e65bab10243987f91ed238a1329f7ab